modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -489,10 +489,9 @@ private Model getParent(Model model) {
 
                 // TODO revise below as we've already performed unaliasing so that the isAlias check may be removed
                 Map<String, Object> modelTemplate = (Map<String, Object>) ((List<Object>) models.get("models")).get(0);
-                // Special handling of aliases only applies to Java
                 if (modelTemplate != null && modelTemplate.containsKey("model")) {
                     CodegenModel m = (CodegenModel) modelTemplate.get("model");
-                    if (m.isAlias) {
+                    if (m.isAlias && !config.shouldGenerateAliasModels()) {
```

Comment:
Instead of using shouldGenerateAliasModels, what about using the option "generateAliasAsModel" instead of forcing the behavior in the generator (as in some use cases, the API owners simply want to reuse the alias for array/map throughout the OpenAPI document)?

Reply:
Do you mean:
- as a command line argument
- or just using the generator boolean property?
i don't know how to do the command line argument option.

Reply:
Please try `--generate-alias-as-model` or `--additional-properties generateAliasAsModel=true`

Reply:
Switched to using the existing `!ModelUtils.isGenerateAliasAsModel()`
which contains a boolean set by invoking the generator with the command line argument `--generate-alias-as-model`